src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6073,4 +6073,25 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureModuleInitializers" xml:space="preserve">
     <value>module initializers</value>
   </data>
+  <data name="ERR_ModuleInitializerMethodMustBeAccessibleOutsideTopLevelType" xml:space="preserve">
+    <value>Module initializer method '{0}' must be accessible outside top-level type '{1}'</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustBeStatic" xml:space="preserve">
+    <value>Module initializer method '{0}' must be static</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeContainedInGenericType" xml:space="preserve">
+    <value>Module initializer method '{0}' must not be contained in a generic type</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeGeneric" xml:space="preserve">
+    <value>Module initializer method '{0}' must not have generic parameters</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotHaveParameters" xml:space="preserve">
+    <value>Module initializer method '{0}' must not have parameters</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustReturnVoid" xml:space="preserve">
+    <value>Module initializer method '{0}' must return void</value>
```

Comment:
Could we collapse several of these diagnostics into one? For instance: "Module initializer method '{0}' must be a static, non-generic method, with no parameters and 'void' return"

Reply:
Sure. I think @RikkiGibson suggested this too. @AlekseyTs I think I'll plan to go ahead with this which will make some of our threads moot.

Reply:
Unified in last push.